### DISCUSSION TITLE
Collapsible Support

### DISCUSSION THREAD

**Discussion by white-parrot** [zim-desktop-wiki/zim-desktop-wiki#2512] [Ideas]
Can we have a Collapsible Support like we have in Markup using details tag. It will be very helpful, to shrink Long articles into few lines, so that page can be more clean and concise.

I am doing it now by using very naïve approach. 

I have updated style.css file to have below Style : 
.wrap-collabsible {
  margin-bottom: 1.2rem 0;
}

input[type='checkbox'] {
  display: none;
}

.lbl-toggle {
  display: block;
  font-weight: bold;
  font-family: monospace;
  font-size: 1.0rem;
  padding: 1rem;
  color: #5294e2;
  cursor: pointer;
  border-radius: 7px;
  transition: all 0.25s ease-out;
}

.lbl-toggle:hover {
  color: #7C5A0B;
}

.lbl-toggle::before {
  content: ' ';
  display: inline-block;

  border-top: 5px solid transparent;
  border-bottom: 5px solid transparent;
  border-left: 5px solid currentColor;
  vertical-align: middle;
  margin-right: .7rem;
  transform: translateY(-2px);

  transition: transform .2s ease-out;
}

.toggle:checked + .lbl-toggle::before {
  transform: rotate(90deg) translateX(-3px);
}

.collapsible-content {
  max-height: 0px;
  overflow: hidden;
  transition: max-height .25s ease-in-out;
}

.toggle:checked + .lbl-toggle + .collapsible-content {
  max-height: 100vh;
}

.toggle:checked + .lbl-toggle {
  border-bottom-right-radius: 0;
  border-bottom-left-radius: 0;
}

.collapsible-content .content-inner {
  background: rgba(250, 224, 66, .2);
  border-bottom: 1px solid rgba(250, 224, 66, .45);
  border-bottom-left-radius: 7px;
  border-bottom-right-radius: 7px;
  padding: .5rem 1rem;
}

and Then adding "\<dummy-collapsible-tag\>" in notebook text like this : 

![image](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/82340454/c8726271-ca60-4eef-92b6-ce48794a3402)

And then Exporting the Notebook as HTML and replacing the string using notepadd++ to get the proper HTML div

![image](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/82340454/06e472bc-c95d-4168-9616-bb828cfaf3fa)

Giving this Final Result

![image](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/82340454/248fb389-004c-4fa4-a7b3-b575bed3af16)

![image](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/82340454/fdaa8f42-ee0b-4e10-850a-26d6d98a51ec)

![image](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/82340454/c26f5bee-7491-40fc-832b-876712a1cbd6)

I hope this Will not be a big code change but definitely it will be very use usefull.

Please Let me know if I can contribute by any mean. My EMAIL is : runtimeturtle@gmail.com

**Comment by lorenrosen** [comment]
I've long wanted something like this.

**Comment by introt** [comment]
Hi! This has been discussed before in #1667, you might find the HTML export template found therein useful: https://github.com/zim-desktop-wiki/zim-desktop-wiki/discussions/1667#discussioncomment-1749706
